packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -384,22 +385,40 @@ export class TreeViewWidget extends TreeViewWelcomeWidget {
 
     handleEnter(event: KeyboardEvent): void {
         super.handleEnter(event);
-        this.tryExecuteCommand();
+        this.tryExecuteCommand(this.findCommands());
     }
 
     handleClickEvent(node: TreeNode, event: React.MouseEvent<HTMLElement>): void {
         super.handleClickEvent(node, event);
-        this.tryExecuteCommand(node);
+        // If clicked on item (not collapsable icon) - execute command or toggle expansion if item has no command
+        if (!event.currentTarget.hasAttribute('data-node-id')) {
+            const commandMap = this.findCommands(node);
+            if (commandMap.size > 0) {
+                this.tryExecuteCommand(commandMap);
+            } else {
+                if (this.isExpandable(node) && !this.hasShiftMask(event) && !this.hasCtrlCmdMask(event)) {
+                    this.model.toggleNodeExpansion(node);
+                }
+            }
+        }
     }
 
-    // execute TreeItem.command if present
-    protected tryExecuteCommand(node?: TreeNode): void {
+    protected findCommands(node?: TreeNode): Map<string, unknown[]> {
+        const commandMap = new Map<string, unknown[]>();
```

Comment:
I imagine this is a breaking change. Should we try to keep the `tryExecuteCommand(node?: TreeNode): void` signature? It would probably just contain `this.tryExecuteCommandMap(this.findCommands(node));`. Alternatively, the breaking change needs to be documented in the changelog.

Reply:
I restored original `tryExecuteCommand` as you suggested. I have not found any usages of `tryExecuteCommand` anywhere, but I agree, there is no point in breaking api for other extensions.